core-bundle/src/Security/Authentication/RememberMe/ExpiringTokenBasedRememberMeServices.php

```diff
@@ -0,0 +1,253 @@
+                    'expires' => DoctrineType::DATETIME,
+                ]
+            )->fetchAll(\PDO::FETCH_OBJ);
+        } catch (\Exception $e) {
+            $rows = [];
+        }
+
+        if (empty($rows)) {
+            throw new TokenNotFoundException('No token found.');
+        }
+
+        return $rows;
+    }
+
+    private function migrateToken($token): string
+    {
+        $tokenValue = base64_encode(random_bytes(64));
+
+        $this->connection->update(
+            'tl_remember_me',
+            ['expires' => (new \DateTime())->add(new \DateInterval('PT60S'))],
+            ['expires' => DoctrineType::DATETIME]
+        );
+
+        $this->insertToken($token->class, $token->username, $token->series, $tokenValue);
+
+        return $tokenValue;
+    }
+
+    /**
+     * Adds a new rememberme token to the database.
+     */
+    private function insertToken(string $class, string $username, string $series, string $tokenValue): void
+    {
+        $this->connection->insert(
+            'tl_remember_me',
+            [
+                'class' => $class,
+                'username' => $username,
+                'series' => $series,
+                'value' => hash_hmac('sha256', $tokenValue, $this->secret),
+                'lastUsed' => new \DateTime(),
+            ],
+            [
+                'class' => \PDO::PARAM_STR,
+                'username' => \PDO::PARAM_STR,
+                'series' => \PDO::PARAM_STR,
+                'value' => \PDO::PARAM_STR,
+                'lastUsed' => DoctrineType::DATETIME,
+            ]
+        );
+    }
+
+    private function findValidToken(array $rows, string $tokenValue)
+    {
+        $tokenValue = hash_hmac('sha256', $tokenValue, $this->secret);
+
+        while ($token = array_shift($rows)) {
+            try {
+                if (!hash_equals($token->value, $tokenValue)) {
+                    throw new CookieTheftException(
+                        'This token was already used. The account is possibly compromised.'
+                    );
+                }
+
+                if ((new \DateTime($token->lastUsed))->getTimestamp() + $this->options['lifetime'] < time()) {
+                    throw new AuthenticationException('The cookie has expired.');
+                }
+
+                return $token;
+            } catch (AuthenticationException $e) {
+                $lastException = $e;
+            }
+        }
+
+        throw $lastException;
+    }
+
+    /**
+     * Creates a rememberme cookie.
+     */
+    private function createCookie(Request $request, string $series, string $tokenValue): Cookie
+    {
+        return new Cookie(
+            $this->options['name'],
+            $this->encodeCookie([$series, $tokenValue]),
+            time() + $this->options['lifetime'],
+            $this->options['path'],
+            $this->options['domain'],
+            $this->options['secure'] ?? $request->isSecure(),
+            $this->options['httponly'],
```

Comment:
As this is our contao specific implementation we might want to hardcode `httponly` to true but I'm also fine to leave it as is.

Reply:
I don't think that would be a good idea, because the firewall still allows for that configuration but it would be ignored… We could only do that when we change and validate the firewall config.